apps/api/src/donations/donations.service.ts

```diff
@@ -423,17 +423,25 @@ export class DonationsService {
   async createManyBankPayments(donationsDto: CreateManyBankPaymentsDto[]) {
     for (const donation of donationsDto) {
       try {
-        //upserting so that if we import the same file again with additional fields the records to be updated accordingly
-        await this.prisma.donation.upsert({
-          where: { extPaymentIntentId: donation.extPaymentIntentId },
-          update: donation,
-          create: donation,
+        //to avoid incrementing vault amount twice we first try to create
+        await this.prisma.donation.create({
+          data: donation,
         })
 
-        //now update the Vault amounts too
+        //and if unique constraint on extPaymentIntentId fails this line will be skipped
         this.vaultService.incrementVaultAmount(donation.targetVaultId, donation.amount)
```

Comment:
I think relying on exceptions during the donation create is not best approach.
Here is my suggestion:

```
await prisma.$transaction(async (tx) => {
      const oldDonation = this.prisma.donation.findUnique(...);
      await this.prisma.donation.upsert({
          where: { extPaymentIntentId: donation.extPaymentIntentId },
          update: donation,
          create: donation,
      if (!oldDonation) {
            this.vaultService.incrementVaultAmount(donation.targetVaultId, donation.amount)
      } else {
            await this.prisma.donation.update({
                   where: { extPaymentIntentId: donation.extPaymentIntentId },
                    data: donation,
           })
   }
} );

```

Reply:
Thanks for pointing me to a better  approach, done now! The code became better and today I learned how prisma interactive transactions work - love them!